### ISSUE TITLE
SqsPointToPointSubscriber creates too many queues, and does not handle messages

### ISSUE THREAD

**Issue by dariusdamalakas** [justeat/JustSaying#79]
Hi

There's an issue with how SqsPointToPointSubscriber is creating queues. Below example should just create 1 queue, and should handle the message. Instead, it creates 3 queues and does not handle a message.

````c#
var queue = "MyQueue";
var publisher = CreateMeABus.InRegion(RegionEndpoint.EUWest1.SystemName)
                .WithSqsPointToPointSubscriber()
                .IntoQueue(queue)
                .WithMessageHandler<Message1>(new Handler1())
                .WithMessageHandler<Message2>(new Handler2())
                .WithSqsMessagePublisher<Message1>(config => config.QueueName = queue)
                .WithSqsMessagePublisher<Message2>(config => config.QueueName = queue);

publisher.StartListening();

publisher.Publish(new Message2());
````

Code creates 3 queues:
* message1-myqueue
* message2-myqueue
* message2-message1-myqueue

SQS subscriber is listening on message1-myqueue and messagee2-message1-myqueue
Message is published on queue message2-myqueue

I see a couple of issues with this behaviour
1. The code does add a prefix to the queue name which is the message name. This is not evident from the fluent interface. I believe the intention here was to create queue per message, so that the bus would listen on each queue.  This is also different behaviour compared to SnsSubscriber, where only a single queue will be created. Perhaps there's a bug in there as well - maybe the intention is to have a queue per message type, and not a single queue with multiple listeners as it is now.
2. Ref what has been said above, I expect this code to create just 1 queue. And not 3.
3. Message is never handled due to a mismatch in queues

**Comment by dariusdamalakas** [comment]
Hi, just to add that we have a fix for this already in place, but that builds up on [another PR](https://github.com/justeat/JustSaying/pull/78), which has not yet been accepted nor declined.

**Comment by payman81** [comment]
Hi @dariusdamalakas  - Sorry for the late reply. I will take a look at this reported bug in the next couple of days. Thanks a lot for the feedback so far.

**Comment by dariusdamalakas** [comment]
@payman81, thanks, no worries. TBH, am not in a position to submit a PR at the moment too, code is not up to the standards. But just to give a quick idea of what the solution we come up - all logic concerning topic and queue name creation was brought into a new IBusNamingConvention interface. The default implementation creates topic names based on message type name, and queue name is what get's pased in IntoQueue() method. Also added a config option to override this, so now it's possible take in control queue and topic naming. 

````c#
justSaying = CreateMeABus
                .InRegion("region")
                .WithNamingStrategy(() => busNamingConvention)
                .WithSqsPointToPointSubscriber()
                .IntoQueue("queue1")
                .WithMessageHandler<Message1>(handlerResolver)

````

**Comment by payman81** [comment]
Hi @dariusdamalakas - There is certainly a bug in point-to-point model with creating queues and generating queue names where there are multiple handlers per SqsSubscriber. 
This has never been a problem for me because I use a separate queue per message.

Not having message names in the name of the queue seems to fix the problem. 
https://github.com/justeat/JustSaying/compare/master...payman81:issue-79

With the change above, your code will create a single queue 'myqueue' that receives both Message1 and Message2. 
Does that solve the issue you were having?

I noticed that in your example above, you're using point-to-point subscriber (SQS publisher-SQS subscriber) as opposed to pub-sub model (SNS publisher - SQS subscriber). The logic for naming queues are different in point-to-point model than in pub-sub model and the change in my spike only impacts point-to-point model.

**Comment by dariusdamalakas** [comment]
Thanks @payman81.  

I do have some concerns with this fix. This logic should not have been here in the first place. SqsPublishEndpointProvider should be handling this.

>>The logic for naming queues are different in point-to-point model than in pub-sub model 
What is the logic for naming queues in point-to-point and pub-sub models? Isn't that just the name of queue as configured when calling 'IntoQueue(..)' in both cases?

**Comment by payman81** [comment]
That's correct. I'll change this so that all decisions to do with naming comes from IPublishEndpointProvider.
I'll keep this issue open until we have a fix.
I was going to pick this up myself unless you have a PR for this?

**Comment by dariusdamalakas** [comment]
Cool, thanks.  

Yes, we do have already fixed this. As noted in my first comment on this thread, we have introduced `.WithNamingStrategy(() => busNamingConvention)`.  This allows to override naming convention, and , of course, there is a default naming convention.

Just one caveat: you have noted that naming queues for pub-sub and point-to-point models is different. I don't actually think is true. I might be wrong, this is what I want to clarify.  My understanding is that currently queue name logic for both pub-sub and point-to-point is the same and it is (pseudo code): `QueueName is empty?  messageTypeName  : QueueName`, where QueueName is the configured name passed when calling `IntoQueue(...)`

**Comment by dariusdamalakas** [comment]
This has been fixed by PR #96.
